endaq/device/schemata/flash_package.xml

```diff
@@ -36,6 +36,12 @@
     <UIntegerElement name="MinFWRev" id="0xF3" multiple="0" mandatory="0"> Minimum firmware version that can receive this update (in case of interface changes)</UIntegerElement>
 	<IntegerElement name="KeySlot" id="0xF4" multiple="0" mandatory="1"> AES Key slot used to encrypt payload (-1 for unencrypted payload)</IntegerElement>
 	<IntegerElement name="PayloadLen" id="0xF5" multiple="0" mandatory="1"> Length of following encrypted payload  of the following format: 1 or more of (uint16 byte offset || uint16 length || [length] bytes of binary data to flash)</IntegerElement>
+    <StringElement name="TargetProcessor" id="0xF6" multiple="0" mandatory="1"> Processor the attached update is intended for. Values include "EFM32GG11B820F2048GL120", "STM32U585AII6", or "ANY" for non-firmware updates</StringElement>
+    <StringElement name="FWRevString" id="0xF7" multiple="0" mandatory="0"> Human readable firmware revision</StringElement>
+    <StringElement name="UpdateNotes" id="0xF8" multiple="0" mandatory="0"> Human readable notes regarding the update</StringElement>
+    <MasterElement name="FutureElements" global="1" id="0xE0" multiple="0" mandatory="0">The device may skip elements inside FutureElements without stopping the update. Any other unrecognized elements will kill the firmware update.
```

Comment:
`<FutureElements>` seems like a bit of a hack; all new elements will end up going in there, possibly indefinitely.  I'm not sure why you would not want to ignore unknown elements. The most likely causes of unknown elements are either a corrupted file or a newer version of the schema. In the latter case, the MinFWRev would stop the update before the unknown elements would be read. In the former case, corrupted files are nearly always bad in other ways, like invalid element lengths. If there was an expected range of element IDs, the bad file would be identified by an out-of-range ID.

Reply:
It would be great if FWRevString could just go into Future and EFM32 devices could benefit from that. As is, the devices will reject any update with FWRevString.
The intent is to use it for optional elements we add in the future

Reply:
What about a different name, like `<ExtendedElements>`? Maybe `<ExtendedData>`, since Element is kind of redundant?

Reply:
Changed to `NonCriticalElements`, as we discussed